registry/session_v2.go

```diff
@@ -95,11 +98,47 @@ func (r *Session) GetV2ImageManifest(ep *Endpoint, imageName, tagName string, au
 		return nil, "", utils.NewHTTPRequestError(fmt.Sprintf("Server error: %d trying to fetch for %s:%s", res.StatusCode, imageName, tagName), res)
 	}
 
-	buf, err := ioutil.ReadAll(res.Body)
+	manifestBytes, err := ioutil.ReadAll(res.Body)
 	if err != nil {
 		return nil, "", fmt.Errorf("Error while reading the http response: %s", err)
 	}
-	return buf, res.Header.Get(DockerDigestHeader), nil
+
+	hdrDigest := digest.Digest(res.Header.Get(DockerDigestHeader))
+	if res.Header.Get(DockerDigestHeader) != "" {
+		// Need to verify the digest.
+
+		sig, err := libtrust.ParsePrettySignature(manifestBytes, "signatures")
```

Comment:
Can we move this to a separate function? I think we can replace this with something from distribution and that will be easier if there is an existing function.